### PR TITLE
#2111 Update docs to reflect Menas to REST API change

### PR DESCRIPTION
**Pull request by jakipatryk**
- Updated deployment section to reflect REST API separation
- Changed Menas config section to use new version configs
- Changed --menas-auth-... to --rest-api-...

### CODE REVIEW

**Comment by lsulak** [comment]
[x] code reviewed 
 [x] pulled

I searched for other occurrences of `menas`, all seem good except 1 potential thing.

**Comment by jakipatryk** [comment]
> * [x]  code reviewed
> * [x]  pulled
> * [ ]  build (not yet, I'm having problems with installation of ruby-related dependencies on M1 Mac)
> 
> I searched for other occurrences of `menas`, all seem good except 1 potential thing.
> 
> -- Update: The build failed, but it's possible that the problem is on my machine. Were you able to build it? It says:
> 
> ` Liquid Exception: Liquid syntax error (line 50): Variable '{{a}' was not properly terminated with regexp: /\}\}/ in menas/target/node/node_modules/npm/node_modules/balanced-match/README.md`
> 
> I have these versions:
> 
> * Bundler 2.3.21
> * ruby 2.6.8p205
> * jekyll 3.8.5

@lsulak I had the same issue, but it is due to the fact that `target` folder is not version controlled, thus when you do checkout to Github pages branch it stays there. You have to delete `target` folders manually  or run `mvn clean` before checking to the branch (I deleted manually).

File: _docs/3.0.0/usage/rest-api-config.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: docs
```
[the remark this answers is not in the document]
Comment (jakipatryk) [reply _docs/3.0.0/usage/rest-api-config.md:2]: I'm not sure how it works under the hood, but what it generates seems reasonable to me.

So the behaviour is that when you select version 2.0.0, you get this page:
2.0.0 usage
there are links pointing to old version Menas docs: 
2.0.0 usage config
2.0.0 usage api

And when you select version 3.0.0, you get this page:
3.0.0 usage
and there are links pointing to new docs (Menas/REST separation):
3.0.0 usage config
3.0.0 usage api

**Comment by lsulak** [comment]
> > * [x]   code reviewed
> > * [x]   pulled
> > * [ ]   build (not yet, I'm having problems with installation of ruby-related dependencies on M1 Mac)
> > 
> > I searched for other occurrences of `menas`, all seem good except 1 potential thing.
> > -- Update: The build failed, but it's possible that the problem is on my machine. Were you able to build it? It says:
> > ` Liquid Exception: Liquid syntax error (line 50): Variable '{{a}' was not properly terminated with regexp: /\}\}/ in menas/target/node/node_modules/npm/node_modules/balanced-match/README.md`
> > I have these versions:
> > 
> > * Bundler 2.3.21
> > * ruby 2.6.8p205
> > * jekyll 3.8.5
> 
> @lsulak I had the same issue, but it is due to the fact that `target` folder is not version controlled, thus when you do checkout to Github pages branch it stays there. You have to delete `target` folders manually or run `mvn clean` before checking to the branch (I deleted manually).

You are right, thank you. Okay, I was able to build it and play around with the UI, it looks like it works. Approved.

**Review by lsulak — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]
LGTM (just read the markup)

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**